api_app/schemas/azuread.json

```diff
@@ -1,23 +1,29 @@
 {
-    "$schema": "http://json-schema.org/draft-07/schema",
-    "$id": "https://github.com/microsoft/AzureTRE/schema/azuread.json",
-    "type": "object",
-    "title": "Azure AD Authorisation Schema",
-    "default": {},
-    "required": [
-        "client_id"
-    ],
-    "properties": {
-        "client_id": {
-            "type": "string",
-            "title": "Application (Client) ID",
-            "description": "The AAD Application Registration ID for the workspace. Use 'auto_create' if you wish TRE to create this."
-        },
-        "client_secret": {
-          "type": "string",
-          "title": "Application (Client) Secret",
-          "description": "The AAD Application Registration secret for the workspace. Leave blank if using `auto_create` above. This value will be stored in the Workspace Key Vault.",
-          "sensitive": true
-      }
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "$id": "https://github.com/microsoft/AzureTRE/schema/azuread.json",
+  "type": "object",
+  "title": "Azure AD Authorisation Schema",
+  "default": {},
+  "required": [
+    "client_id"
+  ],
+  "properties": {
+    "client_id": {
+      "type": "string",
+      "title": "Application (Client) ID",
+      "description": "The AAD Application Registration ID for the workspace. Use 'auto_create' if you wish TRE to create this."
+    },
+    "client_secret": {
+      "type": "string",
+      "title": "Application (Client) Secret",
+      "description": "The AAD Application Registration secret for the workspace. Leave blank if using `auto_create` above. This value will be stored in the Workspace Key Vault.",
+      "sensitive": true
+    },
+    "create_aad_groups": {
```

Comment:
What does it mean to add it here?
Shouldn't this be in the workspace's template schema?

Reply:
Good question, originally the plan was to have multiple workspace templates, and they could be used with different ID provider confuguration, such as Azure AD, AzureAD B2C, Auth0 etc. The base workspace template has become coupled to Azure AD... as the TF is part of the base workspace. Originally I envisaged the ID piece to be a seperate bundle, that would be a dependancy... but we know the challenges around dependancies.

I don't want to deal with it in this PR, but we need to decide, 1. Is the base workspace become an "AzureAD base workspace"  - I think it is at present, 2. Do we need to seperate out the App Reg creation terraform into a seperate stage. Until we have a firm request (lots talk about it) about using an ID provider other than Azure AD we could move this all into the base workspace template, and remove this json file and associated code.

Maybe create another issue?

Reply:
I've removed from here an put into the workspace. The properties get merged anyway.